src/txn/txn_recover.c

```diff
@@ -959,6 +973,14 @@ __wt_txn_recover(WT_SESSION_IMPL *session, const char *cfg[])
 done:
     /* Close cached cursors, rollback-to-stable asserts exclusive access. */
     WT_ERR(__recovery_close_cursors(&r));
+#ifndef WT_STANDALONE_BUILD
+    if (conn->unclean_shutdown && conn->recovery_major == 10 && conn->recovery_minor == 0 &&
+      conn->recovery_patch == 0) {
+        WT_ERR_MSG(session, WT_ERROR,
+          "Upgrading a version on an unclean shutdown database is not allowed. Perform a clean "
```

Comment:
"unclean shutdown database" should just be:  "unclean database" or "database that was not shutdown cleanly"

Reply:
Thanks fixed as suggested with "database that was not shutdown cleanly".